### ISSUE TITLE
Extract all imports not working

### ISSUE THREAD

**Issue by Frotty** [peq/WurstScript#438]
Your commit only changed the exception.
Please test this

````
Mär 28, 2016 1:46:17 AM de.peeeq.wurstscript.WLoggerDefault severe
SEVERE: Error
java.lang.RuntimeException: java.lang.ArrayIndexOutOfBoundsException: 24904
	at de.peeeq.wurstio.map.importer.ImportFile.extractImportedFiles(ImportFile.java:71)
	at de.peeeq.wurstio.map.importer.ImportFile.extractImportedFiles(ImportFile.java:142)
	at de.peeeq.wurstio.Main.main(Main.java:89)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 24904
	at systems.crigges.jmpq3.MpqCrypto.hash(MpqCrypto.java:90)
	at systems.crigges.jmpq3.HashTable.getBlockIndexOfFile(HashTable.java:104)
	at systems.crigges.jmpq3.JMpqEditor.extractFile(JMpqEditor.java:297)
	at de.peeeq.wurstio.mpq.Jmpq3BasedEditor.extractFile(Jmpq3BasedEditor.java:45)
	at de.peeeq.wurstio.map.importer.ImportFile.extractImportedFiles(ImportFile.java:54)
	... 2 more
````

**HOW TO REPRODUCE**

1. Creaty empty map
2. Place bloodmage to make it w3x
3. import any file
4. save map, save again after warning
5. Extract all imports
6. Get raped by Exception

**Comment by Crigges** [comment]
Well i checked it out and cannot explain what is going wrong. 
My knowlege that far:
-The MpqCrypto.hash(...) function is trying to generate a hash for the file that needs to be extraced
-For some reason this function accesses the crypttable at a wrong position which causes the out of bounds exception
@peq ported the MpqCrypto code from storm lib so he may be able to explain what is going wrong.

**Comment by Frotty** [comment]
fixed by cfeb97371a62d8c2821cbe81741fde423cbe0a59
